File: LeetCode/Python/Medium/647-Palindromic-Substrings/solution.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        result=0

        n=len(s)

        for index in range(0, n):
            result+=self.expand_around_center(s, n, index, index)

            result+=self.expand_around_center(s, n, index, index + 1)

        return result

    def expand_around_center(self, s: str, n: int, left: int, right: int) -> int:
        palindromes=0

        while left >= 0 and right < n and s[left]==s[right]:
            palindromes+=1

            left-=1

            right+=1

        return palindromes

File: LeetCode/Rust/Medium/647-Palindromic-Substrings/solution.rs
impl Solution
{
    pub fn count_substrings(s: String) -> i32
    {
        let mut result: i32=0;

        let n: usize=s.len();

        let characters: Vec<char>=s.chars().collect::<Vec<char>>();

        for index in 0..n
        {
            Self::expand_around_center(&characters, &mut result, n, index as i32, index);

            Self::expand_around_center(&characters, &mut result, n, index as i32, index + 1);
        }

        result
    }

    fn expand_around_center(characters: &Vec<char>, result: &mut i32, n: usize, mut left: i32, mut right: usize) -> ()
    {
        while left >= 0 && right < n && characters[left as usize]==characters[right]
        {
            *result+=1;
            
            left-=1;

            right+=1;
        }
    }
}